src/melee.cpp

```diff
@@ -2135,7 +2209,7 @@ std::string Character::melee_special_effects( Creature &t, damage_instance &d, i
     std::string target = t.disp_name();
 
     if( has_active_bionic( bio_shock ) && get_power_level() >= bio_shock->power_trigger &&
-        ( !is_armed() || weapon.conductive() ) ) {
+        ( weap.is_null() || weapon.conductive() ) ) {
```

Comment:
Aren't `weap` and `weapon` the same in the end?

Reply:
Unless I misread `weapon` refers back to the character's wielded weapon, whereas `weap` is the argument passed to this function. `attack_overwrite` techs instead pass null as their `weap`. 

I tried setting `cur_weapon` to null before which would have been much cleaner but it made the player's weapon disappear on a trigger.

Reply:
When `melee_special_effects` is called from `melee_attack_abstract`, `weap` parameter is `cur_weapon`.

Then `item *cur_weapon = allow_unarmed ? &used_weapon() : &weapon;` and `used_weapon` is defined as `martial_arts_data->selected_force_unarmed() ? null_item_reference() : weapon;`

Basically if unarmed attack is allowed, currently wielded weapon (`weapon`) will bed used unless martial art forces an unarmed attack.

What bothers me is `melee_special_effects` using both `weap` and `weapon` in some checks. Say, you attack w/o weapon (because martial art forces you) and `weap` is `null_item_reference`, but you will still get zapped in case your `weapon` is conductive and you will add fire damage if wielded `weapon` has `FLAMING` flag.

Reply:
Yes, the current sequence of the attack is weird like that. Ideally I'd imagine changing the sequence so that the attempted technique gets decided *first* setting the used weapon (forced unarmed/attack override or your wielded weapon), it uses that weapons' to-hit for the hit roll, *then* checking if the technique you were attempting succeeded (and applying the message/special effects/monster defense depending on what weapon you used). However, while that would make more sense it would also pretty much completely invalidate the current martial arts' sequencing of techniques and general balance - that isn't something I feel comfortable doing, especially not without having more of a plan for MAs than "it'll be neater under the hood".